packages/java/endpoint/src/test/java/com/vaadin/hilla/ExplicitNullableTypeCheckerTest.java

```diff
@@ -500,7 +500,7 @@ public Long methodWithIdAnnotation() {
     /**
      * Method for testing
      */
-    @Nonnull
+    @javax.annotation.Nonnull
     public String stringNonnull() {
         return "";
     }
```

Comment:
javax.annotation.Nonnull is different than other NonNull annotations in the sense that it can be found on the method even when it is applied to the return type like here. All other NonNull annotations when applied on the return type are only returned for the return type and not the method. Hence this test will fail if the annotation is changed to org.jspecify

Reply:
This was briefly discussed in Slack also without any conclusion about what the test actually tests or how it should be fixed, if at all. So I left it like it was for now

Reply:
It might be better to ask my question this way:
The JSpecify's NonNull is targeted as `@Target(TYPE_USE)`, while our deprecated Nonnull was defined to target like this: `@Target({ ElementType.TYPE_USE })`. That makes a difference when the `AnnotatedElement#getAnnotations` is called [here](https://github.com/vaadin/hilla/blob/main/packages/java/endpoint/src/main/java/com/vaadin/hilla/ExplicitNullableTypeChecker.java#L57). Shouldn't this difference be considered in the `isRequired` implementation?

Reply:
The test also fails if you replace `javax.annotation.Nonnull` with the old `com.vaadin.hilla.Nonnull`

Reply:
So I don't think it is relevant for this PR